### ISSUE TITLE
ipywidget background doesn't respect theme

### ISSUE THREAD

**Issue by rebornix** [microsoft/vscode-jupyter#7161]
## Environment data

-   VS Code version: 1.60-Insiders
-   Jupyter Extension version (available under the Extensions sidebar): 2021.8.2027072902

![image](https://user-images.githubusercontent.com/876920/129631201-cbc4c754-1a42-4375-bd89-aaae810ed647.png)

**Comment by DonJayamanne** [comment]
We hardcode the background, as IPYWidgets doesn't understand VS Code themes.
& it defaults to white backgrounds in jupyter & jupyter lab, 
If we want to fix this, we need to get ipywidgets & all of jupyter widgets to understand vscode themes (fonts, etc).

**Comment by UncleHoo** [comment]
In the meantime, is it possible to change the background colour within the Jupyter Cell itself?  It's very hard on the eyes when VSCode by default is black (such is my install), and then you get a large white cell on the screen.  I understand that it would be a lot of work to get the ipyWidgets to understand and respect VSCode themes, but having the ability to manually set the background colour would really help and I think would be much easier to implement?

**Comment by DonJayamanne** [comment]
@UncleHoo setting background color the way you suggest will not work always, as the foreground color of the text would not match the background and some of the text might no longer be legible.
Unfortunately I don't see an easy way around this that would make it work for all scenarios. We can't have it working for a few and not others (unusable).

However if someone believes this can be done we're happy to take a PR

**Comment by martinRenou** [comment]
I believe this can be achieved by making a bridge from the vscode theme (CSS variables?) to the jupyterlab theme.

ipywidgets (and other libraries for Jupyter) use the JupyterLab CSS variables for respecting the theme.

It might just be a matter of defining the needed CSS variables using the current VSCode theme.

**Comment by chandradharkoneti** [comment]
> ## Environment data
> 
>     * VS Code version: 1.60-Insiders
> 
>     * Jupyter Extension version (available under the Extensions sidebar): 2021.8.2027072902
> 
> 
> ![image](https://user-images.githubusercontent.com/876920/129631201-cbc4c754-1a42-4375-bd89-aaae810ed647.png)

Hi, Can you please tell me how to access that option ```.cell-output-ipywidget-background{}``` so that I can change it for myself and see if it works.

**Comment by DonJayamanne** [comment]
> e how to access that option .cell-output-ipywidget-background{} so that I can change it for myself and see if it works.

Currently there's no easy way to do this, we just need to add support for the themes

**Comment by DonJayamanne** [comment]
> I believe this can be achieved by making a bridge from the vscode theme (CSS variables?) to the jupyterlab theme.

Thanks, thats the plan, i think what I wrote earlier makes it sound like we'll need to modify `ipywidgets` upstream to support vscode, but that's not what i mean, will update acordingly.

**Comment by chandradharkoneti** [comment]
**For anyone interested:**
**I have found a hacky sacky solution.**


Make sure VS Code is closed(at least no jupyter related things should be running on it[ I personally closed it completely and reopened it for editing the below file]). 

Open file explorer, go to:

```C:\Users\<your_username_here>\.vscode\extensions\ms-toolsai.jupyter-2022.4.1021342353\out\webviews\webview-side\ipywidgetsRenderer```

Open the **ipywidgetsRenderer.js** file (using notepad++ or notepad or even VSCode(before letting any jupyter stuff run)) and 

wherever you see ```cell-output-ipywidget-background``` followed by ```background: white```, replace ```white``` with ```black```.

Now the problem is solved. Enjoy :)

**Comment by yusisc** [comment]
> 
@chandradharkoneti 
This method works good. But, the ***text color*** of additional information is black, too.  
Is there a way to make the texts be ***brighter*** ?


![2022-11-15_235300](https://user-images.githubusercontent.com/23088139/201965097-ae99c05c-719c-48bb-bd55-8f6700cf197c.png)

**Comment by starball5** [comment]
Related Stack Overflow question: [How can I get rid of or recolour the background of ipywidgets frames in a Jupyter Notebook in VS Code?](https://stackoverflow.com/q/75477316/11107541)

Other issue tickets that look related: #10609 and #9403

**Comment by npulidomateo** [comment]
> wherever you see `cell-output-ipywidget-background` followed by `background: white`, replace `white` with `black`.

`tansparent` also works. Nice if you like to change to a light color scheme in less dark situations.

**Comment by kylemcdonald** [comment]
> @chandradharkoneti This method works good. But, the _**text color**_ of additional information is black, too. Is there a way to make the texts be _**brighter**_ ?

Here is a hack for making the print statements readable:

```python
css_widget = widgets.HTML("""
<style>
.cell-output-ipywidget-background {
   background-color: transparent !important;
}
.cell-output-ipywidget-background pre {
   color: white !important;
}
</style>
""")
display(css_widget)
```

**Comment by Shisoik** [comment]
I added following HTML cell that makes background transparent and binds widgets' foreground text colour and size to that of the active VSCode editor theme.

```html
%%html
<style>
.cell-output-ipywidget-background {
    background-color: transparent !important;
}
:root {
    --jp-widgets-color: var(--vscode-editor-foreground);
    --jp-widgets-font-size: var(--vscode-editor-font-size);
}  
</style>
```

**Comment by tianfanghan** [comment]
Is there a way to just get rid of the frame it self? Just set margin or padding or box to zero?

**Comment by alex180500** [comment]
> Is there a way to just get rid of the frame it self? Just set margin or padding or box to zero?

Yes I would really like this!

**Comment by Jnorm911** [comment]
bump.

**Comment by flying-sheep** [comment]
I think the jupyter team should define some mandatory CSS variables that are defined in any frontend, including Jupyter Lab, VS Code, and the output of HTML conversion scripts (e.g. nbconvert, MyST-nb)

**Comment by martinRenou** [comment]
It's actually already defined. ipywidgets requires the same CSS variables as a [JupyterLab theme would provide](https://github.com/jupyterlab/extension-cookiecutter-ts/blob/4.0/%7B%7Bcookiecutter.python_name%7D%7D/style/variables.css).

Note that ipywidgets provides a set of default values for when there is no JupyterLab theme available in the page: https://github.com/jupyter-widgets/ipywidgets/blob/main/packages/controls/css/labvariables.css. The VSCode jupyter extension should make a bridge between the current VSCode theme and the Lab variables.

**Comment by flying-sheep** [comment]
I see, so this is really just a missing feature in this extension.

The extension should define all these variables similar to how it’s done in this: https://github.com/microsoft/vscode-jupyter/issues/7161#issuecomment-1616627670

**Comment by alimantu** [comment]
I would love to see this one available in a newer version. 

@DonJayamanne, how can I support this feature so it goes into development?

**Comment by bjornasm** [comment]
Is there any progress here?

**Comment by kevinjohncutler** [comment]
Would love to see this too

**Comment by DonJayamanne** [comment]
I've tried to get this working, and found that it only works partially.
VS Code uses the old IPyWidget architecture, and old Jupyter ipywidget architecture doesn't support theming.
The ipywidgets code in Juypyter lab (is based on a new architecture and scripts/css), which supports theming.
However at this stage we have no plans to support that model as its a massive undertaking with little benefit.

TLDR:
* Theming can be added but only buttons, sliders and basic widgets will support theming
* Plots like beakerx will not be themed in VS Code (it only works when using Jupyter Lab, even Jupyter Notebooks (old application) didn't support theming, which is what VS Code uses)

**Comment by flying-sheep** [comment]
Actually being able to read text consistently is “very little benefit”?
